clients/src/main/java/org/apache/kafka/clients/consumer/StickyAssignor.java

```diff
@@ -216,6 +216,8 @@ public ByteBuffer subscriptionUserData(Set<String> topics) {
 
     @Override
     protected MemberData memberData(Subscription subscription) {
+        // Always deserialize ownedPartitions and generation id from user data
+        // since StickyAssignor is an eager rebalance protocol that will revoke all existing partitions before joining group
```

Comment:
In StickyAssignor, we still deserialize the user data to retrieve all the data since it is an eager rebalance protocol.